src/core/network/qgshttpheaders.h

```diff
@@ -0,0 +1,109 @@
+/***************************************************************************
+                       qgshttpheaders.h
+  This class implements simple http header management.
+
+                              -------------------
+          begin                : 2021-09-09
+          copyright            : (C) 2021 B. De Mezzo
+          email                : benoit dot de dot mezzo at oslandia dot com
+
+***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSHTTPHEADERS_H
+#define QGSHTTPHEADERS_H
+
+#include <QNetworkRequest>
+#include <QMap>
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include "qgssettingsentry.h"
+
+/**
+ * \ingroup core
+ * \brief This class implements simple http header management.
+ * \since QGIS 3.22
+ */
+class CORE_EXPORT QgsHttpHeaders
+{
+  public:
+
+    /**
+     * used in settings
+     */
+    static const QString KEY_PREFIX;
+
+    /**
+     * \brief constructor from map
+     * \param headers
+     */
+    QgsHttpHeaders( const QMap<QString, QVariant> &headers );
+
+    /**
+     * \brief default constructor
+     */
+    QgsHttpHeaders();
+
+    /**
+     * \brief constructor from \a QgsSettings
+     * \param settings
+     * \param key
+     */
+    QgsHttpHeaders( const QgsSettings &settings, const QString &key = QString() );
+
+    virtual ~QgsHttpHeaders();
+
+    /**
+     * \brief update the \a request by adding all the http headers
+     * \param request
+     * \return true if the update succeed
```

Comment:
```suggestion
     * \brief Updates a \a request by adding all the HTTP headers
     * \return TRUE if the update succeed
```

Docs should be sentence case, using "Updates"/"Sets"/"Returns" etc instead of "Update"/"Return" etc.

Use "TRUE" instead of "true" (and "FALSE", "NULLPTR") so that the Python documents automatically convert these to True/False/None

Reply:
ok thanks